pkg/mutate/webhook_test.go

```diff
@@ -0,0 +1,617 @@
+package mutate
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+)
+
+func TestWebhookManager(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Webhook Suite")
+}
+
+var _ = Describe("Webhook", func() {
+
+	var (
+		virtualNodeToleration = corev1.Toleration{
+			Key:      liqoconst.VirtualNodeTolerationKey,
+			Operator: corev1.TolerationOpExists,
+			Effect:   corev1.TaintEffectNoExecute,
+		}
+	)
+
+	Context("1 - Check the new created toleration according to the PodOffloadingStrategy", func() {
+		emptyToleration := corev1.Toleration{}
+		DescribeTable("3 Different type of PodOffloadingStrategy",
+			func(strategy offv1alpha1.PodOffloadingStrategyType, expectedToleration corev1.Toleration) {
+				By(fmt.Sprintf("Testing %s", strategy))
+				Expect(createTolerationFromNamespaceOffloading(strategy)).To(Equal(expectedToleration))
+			},
+			Entry("LocalPodOffloadingStrategyType", offv1alpha1.LocalPodOffloadingStrategyType, emptyToleration),
+			Entry("RemotePodOffloadingStrategyType", offv1alpha1.RemotePodOffloadingStrategyType, virtualNodeToleration),
+			Entry("LocalAndRemotePodOffloadingStrategyType", offv1alpha1.LocalAndRemotePodOffloadingStrategyType, virtualNodeToleration),
+		)
+	})
+
+	Context("2 - Check the NodeSelector imposed by the NamespaceOffloading", func() {
+		// slice with 3 namespaceOffloading one for each PodOffloadingStrategy
+		namespaceOffloadings := []*offv1alpha1.NamespaceOffloading{{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "offloading",
+				Namespace: "test",
+			},
+			Spec: offv1alpha1.NamespaceOffloadingSpec{
+				NamespaceMappingStrategy: offv1alpha1.EnforceSameNameMappingStrategyType,
+				PodOffloadingStrategy:    offv1alpha1.LocalPodOffloadingStrategyType,
+				ClusterSelector: corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"A,B"},
+						}, {
+							Key:      "provider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"C,D"},
+						}, {
+							Key:      "NotProvider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+				},
+				},
+			},
```

Comment:
It seems to me this is copyed and pasted three times, while changing the `PodOffloadingStrategy` only. You should use a function to generate the three items, avoiding code duplication and highlighting the differences.